turbopack/crates/turbo-tasks-backend/src/database/db_invalidation.rs

```diff
@@ -0,0 +1,54 @@
+use std::{
+    fs::{self, read_dir},
+    io,
+    path::Path,
+};
+
+const INVALIDATION_MARKER: &str = "__turbo_tasks_invalidated_db";
+
+/// Atomically write an invalidation marker.
+///
+/// Because attempting to delete currently open database files could cause issues, actual deletion
+/// of files is deferred until the next start-up (in [`check_db_invalidation`]).
+///
+/// In the case that no database is currently open (e.g. via a separate CLI subcommand), you should
+/// call [`cleanup_db`] after this to eagerly remove the database files.
+///
+/// This should be run with the base (non-versioned) path, as that likely aligns closest with user
+/// expectations (e.g. if they're clearing the cache for disk space reasons).
+pub fn invalidate_db(base_path: &Path) -> io::Result<()> {
+    fs::write(base_path.join(INVALIDATION_MARKER), [0u8; 0])?;
+    Ok(())
+}
+
+/// Called during startup. See if the db is in a partially-completed invalidation state. Find and
+/// delete any invalidated database files.
+///
+/// This should be run with the base (non-versioned) path.
+pub fn check_db_invalidation_and_cleanup(base_path: &Path) -> io::Result<()> {
+    if fs::exists(base_path.join(INVALIDATION_MARKER))? {
+        // if this cleanup fails, we might try to open an invalid database later, so it's best to
+        // just propagate the error here.
+        cleanup_db(base_path)?;
```

Comment:
```suggestion
        cleanup_db(base_path).with_context(||format!("Unable to remove invalid database. If this issue persists you can work around by deleting: {base_path}.")?;
```

What do you think about a message like that?  This failing seems pretty fatal